src/EditorFeatures/CSharpTest/RefactoringHelpers/RefactoringHelpersTests.cs

```diff
@@ -0,0 +1,750 @@
+    [|void M()
+    {
+    }|]|}
+}";
+            await TestAsync<MethodDeclarationSyntax>(testText);
+        }
+
+        #endregion
+
+        #region Extractions general
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestExtractionsClimbing()
+        {
+            var testText = @"
+using System;
+class C
+{
+    void M()
+    {
+        var a = {|result:new object()|};[||]
+    }
+}";
+            await TestAsync<ObjectCreationExpressionSyntax>(testText);
+        }
+
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestMissingExtractHeaderForSelection()
+        {
+            var testText = @"
+using System;
+class C
+{
+    class TestAttribute : Attribute { }
+    [Test] public [|int|] a { get; set; }
+}";
+            await TestMissingAsync<PropertyDeclarationSyntax>(testText);
+        }
+        #endregion
+
+        #region Extractions
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestExtractFromDeclaration()
+        {
+            var testText = @"
+using System;
+class C
+{
+    void M()
+    {
+        [|var a = {|result:new object()|};|]
+    }
+}";
+            await TestAsync<ObjectCreationExpressionSyntax>(testText);
+        }
+
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestExtractFromDeclaration2()
+        {
+            var testText = @"
+using System;
+class C
+{
+    void M()
+    {
+        var a = [|{|result:new object()|};|]
+    }
+}";
+            await TestAsync<ObjectCreationExpressionSyntax>(testText);
+        }
+
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestExtractFromAssignment()
+        {
+            var testText = @"
+using System;
+class C
+{
+    void M()
+    {
+        object a;
+        a = [|{|result:new object()|};|]
+    }
+}";
+            await TestAsync<ObjectCreationExpressionSyntax>(testText);
+        }
+
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestExtractInHeaderOfProperty()
+        {
+            var testText = @"
+using System;
+class C
+{
+    class TestAttribute : Attribute { }
+    {|result:[Test] public i[||]nt a { get; set; }|}
+}";
+            await TestAsync<PropertyDeclarationSyntax>(testText);
+        }
+
+        [Fact]
+        [WorkItem(35525, "https://github.com/dotnet/roslyn/issues/35525")]
+        public async Task TestMissingExtractNotInHeaderOfProperty()
+        {
+            var testText = @"
+using System;
+class C
+{
+    class TestAttribute : Attribute { }
+    [Test] public int a { [||]get; set; }
+}";
+            await TestMissingAsync<PropertyDeclarationSyntax>(testText);
+        }
+
+        #endregion
```

Comment:
i didn't actually validate the tests.  but i'm very happy we have them like this.